File: frontend/src/app/models/Bundle.ts
export class Bundle {
    constructor (
        public id : number =0 ,
        public  name : string = "", 
        public bundleType : string = "",     
        public notes : string = "",       
        public providerID : number = 0,
        public price : number = 0,
        public isActive : boolean = true,
        public buyingPrice:number=0

    ){}

}
export class GetBundle {
    constructor (
        public id : number =0 ,
        public  name : string = "", 
        public bundleType : string = "",     
        public notes : string = "",       
        public providerName : string = "",
        public price : number = 0,
        public isActive : boolean = true,
        public buyingPrice:number=0

    ){}

}

File: backend/ISP.DAL/Models/Bundle.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ISP.DAL.Models
{
    public class Bundle
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string BundleType { get; set; } = string.Empty;
        public string Notes { get; set; } = string.Empty;

        [ForeignKey("provider")]
        public int ProviderID { get; set; }
        public InternetServiceProvider provider { get; set; }

        [Required]
        public double Price { get; set; }

        [Required]
        public bool IsActive { get; set; }

        [Required]
        public double BuyingPrice { get; set; }
    }
}
